rdf/common/src/main/scala/org/w3/banana/io/AnyRDFReader.scala

```diff
@@ -0,0 +1,12 @@
+package org.w3.banana
+package io
+
+
+/** RDF reader. */
+trait AnyRDFReader[Rdf <: RDF, M[_]] {
+  
+	/** Read triples from the given location. The syntax is determined from input source URI
+   *  (content negotiation or extension). */
+   def readAnyRDFSyntax( uri:String) : M[Rdf#Graph]
```

Comment:
As mentioned by @bblfish that stuff should go in `io`, at least for now.

I personally don't like the name `AnyRDFReader`. It has nothing to do with `RDFReader`. Also `uri` should at least be `url`. Let's get this type right as well (I know it's not the case for other abstractions yet): either `java.net.URL` (it's ok for scala-js) or `Rdf#URI`.

Here is a proposal:

```scala
trait RDFLoader[Rdf <: RDF, M[_]] {
   def load(url: java.net.URL) : M[Rdf#Graph]
}
```